drivers/scheduler/k8s/k8s.go

```diff
@@ -4328,8 +4328,40 @@ func (k *K8s) GetVolumes(ctx *scheduler.Context) ([]*volume.Volume, error) {
 				// check if the pvc has our VM as the owner
 				want := false
 				for _, ownerRef := range pvc.OwnerReferences {
-					if ownerRef.Kind == vm.Kind && ownerRef.Name == vm.Name {
-						want = true
+					if ownerRef.Kind == vm.Kind {
+						if ownerRef.Name == vm.Name {
+							want = true
+						}
+					}
+					// If volume is still not found, try the datavolumes
+					if !want && ownerRef.Kind == "DataVolume" {
+						// get datavolume
+						kvClient := k8sKubevirt.GetKubevirtClient()
+						dvList, err := kvClient.CdiClient().CdiV1beta1().DataVolumes(vm.Namespace).List(context.TODO(), metav1.ListOptions{})
+						if err != nil {
+							return nil, fmt.Errorf("error listing datavolumes in namespace: %s", vm.Namespace)
+						}
+						for _, dataVol := range dvList.Items {
+							for _, ownerRef := range dataVol.OwnerReferences {
+								if ownerRef.Name == vm.Name {
+									want = true
+								}
+							}
+						}
+					}
+				}
+				// If volume is still not found, try the virt-launcher pod as well
+				if !want {
+					virtLauncherPodsList, err := k8sCore.GetPods(vm.Namespace, nil)
```

Comment:
Similar comment here. We should only consider the pods that are using this PVC. Skip the other pods.

Reply:
No change needed here since we do need to list all pods in the namespace.